### PR TITLE
refactor: Merge new expression

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
@mergify update

**Comment by Xuanwo** [comment]
Conflicts should be resolved. I tried locally but can't compile by a simple merge.

File: tests/sqllogictests/suites/query/02_function/02_0012_function_datetimes [hunk 1]
```diff
@@ -42,7 +42,7 @@ select to_datetime(1630320462000000), to_int64(to_datetime(1630320462000000))  =
 2021-08-30 10:47:42.000000 1
 
 query TB
-select to_date(18869), to_uint32(to_date(18869))  = 18869
+select to_date(18869), to_int64(to_date(18869))  = 18869
```
Comment (TCeason) [inline tests/sqllogictests/suites/query/02_function/02_0012_function_datetimes:45]: There I think we can replase to use to_int64

File: src/query/service/src/interpreters/interpreter_table_recluster.rs [hunk 2]
```diff
@@ -53,26 +52,25 @@ impl Interpreter for ReclusterTableInterpreter {
...
+            Some(PushDownInfo {
+                filters: vec![filter],
+                ..PushDownInfo::default()
+            })
+        } else {
+            None
         };
 
         loop {
-            let table = self
```
Comment (zhyass) [inline src/query/service/src/interpreters/interpreter_table_recluster.rs:70]: Here it is refreshing the table meta and cannot be deleted.

Comment (sundy-li) [reply src/query/service/src/interpreters/interpreter_table_recluster.rs]: thanks! fixed

**Pull request merged**